index.bs

```diff
@@ -3554,15 +3554,68 @@ To <dfn>get the base network event data</dfn> given |request| and |redirect coun
 
 </div>
 
+#### The network.BytesValue Type #### {#type-network-BytesValue}
+
+<pre class="cddl local-cddl remote-cddl">
+network.BytesValue = network.StringValue / network.Base64Value;
+
+network.StringValue = {
+  type: "string",
+  value: text,
+}
+
+network.Base64Value = {
+  type: "base64",
+  value: text,
+}
+</pre>
+
+The <code>network.BytesValue</code> type represents binary data sent over the
+network. Valid UTF-8 is represented with the <code>network.StringValue</code>
+type, any other data is represented in Base64-encoded form as
+<code>network.Base64Body</code>.
+
+<div algorithm>
+To <dfn ignore>deserialize protocol bytes</dfn> given |protocol bytes|:
+
+Note: this takes bytes encoded as a <code>network.BytesValue</code> and returns
+a [=byte sequence=].
+
+1. If |protocol bytes| matches the <code>network.StringValue</code> production,
+   let |bytes| be [=UTF-8 encode=] |protocol bytes|["<code>value</code>"].
+
+1. Otherwise if |protocol bytes| matches the <code>network.Base64Value</code>
+   production. Let |bytes| be [=forgiving-base64 decode=] |protocol
+   bytes|["<code>value</code>"].
+
+1. Return |bytes|.
+
+</div>
+
+<div algorithm>
+To <dfn>serialize protocol bytes</dfn> given |bytes|:
+
+Note: this takes a [=byte sequence=] and returns a <code>network.BytesValue</code>.
+
+1. If |bytes| matches the <code>network.StringValue</code> production,
+   let |protocol value| be [=UTF-8 encode=] |bytes|["<code>value</code>"].
+
+1. Otherwise if |bytes| matches the <code>network.Base64Value</code>
+   production. Let |protocol value| be [=forgiving-base64 decode=]
+   |bytes|["<code>value</code>"].
+
+1. Return |protocol value|.
+
+</div>
+
 #### The network.Cookie Type #### {#type-network-Cookie}
 
 [=Remote end definition=] and [=local end definition=]
 
 <pre class="cddl local-cddl">
 network.Cookie = {
     name: text,
-    ? value: text,
-    ? binaryValue: [ uint ]
+    value: network.BytesValue,
```

Comment:
Does this mean that value will always be present now? It's no longer optional for a remote end to send it?